heartbeat/updater.go

```diff
@@ -0,0 +1,53 @@
+package heartbeat
+
+import "time"
+
+// An Updater represents a function called at each Heart's update interval. It
+// is responsible for preforming any action the heart should preform at the
+// aforementioned interval.
+//
+// If an error is encountered while processing, it may be returned, and the
+// execution of the Updater may be halted.
+type Updater func(h *Heart, now time.Time) error
+
+const (
+	// ISO8601TimeFormat is a time format string according to the time
+	// package and is used to marshal time.Time instances into
+	// ISO8601-compliant strings.
+	//
+	// See: https://www.ietf.org/rfc/rfc3339.txt for more details.
+	ISO8601TimeFormat string = "2006-01-02T15:04:05"
+)
+
+var (
+	// HashUpdateTicker is a default implementation of the Updater type. It
+	// assumes the HASH type is used in Redis to store heartbeat updates
+	// with the following structure:
+	//
+	// ```
+	// 0) ...
+	// 1) HEARTBEAT_ID -> LAST_UPDATE_TIME
+	// 2) ...
+	// 3) ...
+	// ```
+	//
+	// The contents of the HASH are stored at the Heart's Keyspace(), and
+	// the individual HEARTBEAT_IDs are stored at the Heart's Name().
+	//
+	// The `LAST_UPDATE_TIME` is stored as an ISO8601 DateTime, according
+	// to the `ISO8601TimeFormat` constant above.
+	HashUpdateTicker Updater = func(h *Heart, now time.Time) error {
+		redis := h.Pool().Get()
+		defer redis.Close()
+
+		formatted := now.Format(ISO8601TimeFormat)
+
+		if _, err := redis.Do("HSET", h.Location().Keyspace(),
+			h.Location().Name(), formatted); err != nil {
+
+			return err
+		}
+
+		return nil
+	}
```

Comment:
I definitely see the problem here, but not sure that I've found the solution yet. I maintain that the two types should remain separate, but the configuration parameters for each are strikingly similar. The first solution that I thought of was to unify the parameters and have a constructor fn that returns both types, the Heart and the Detector.

I am not sure if this is the right solution, but it seems to address the problem of matching configurations and keeping the two types separate.

If you think this works, I'll go ahead and implement it.